cg/apps/housekeeper/hk.py

```diff
@@ -185,6 +185,19 @@ def get_latest_file(
         files: Query = self._store.get_files(bundle_name=bundle, tag_names=tags, version_id=version)
         return files.order_by(File.id.desc()).first()
 
+    def get_file_by_exact_tags(
+        self, bundle: str, tags: list, version: int | None = None
+    ) -> File | None:
+        """Return a file that matches exact tags and optionally filtered by bundle and version."""
+        if not tags:
+            LOG.debug("No tags provided, skipping")
+            return None
+        files: Query = self._store.get_files(bundle_name=bundle, version_id=version, tag_names=tags)
```

Comment:
This function defined in the API can do that without calling the `_store`
```suggestion
        files: Query = self.files(bundle_name=bundle, version_id=version, tags=tags)
```

Reply:
thanks, added